### PR TITLE
Documentation for Metadata Routing in ASP.NET Core OData 8

### PR DESCRIPTION
**Pull request by gathogojr**
Documentation for Metadata Routing in ASP.NET Core OData 8

### CODE REVIEW

File: Odata-docs/webapi-8/fundamentals/metadata-routing.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+```http
+http://localhost:5000/odata/Orders
+```
+
+## OData service metadata document
+The service metadata document contains a complete description of the feeds, types, properties, and relationships exposed by the OData service in the Entity Data Model (EDM).
+
+A service metadata document describes it's data in EDM terms using an XML language for describing models called the Conceptual Schema Definition Language (CSDL). When exposed by an OData service as a service metadata document, the CSDL document is packed using the EDMX format.
+
+Most OData clients libraries use the service metadata to drive the generation of client-side proxy classes to represent the server types.
```
Comment (habbes) [inline Odata-docs/webapi-8/fundamentals/metadata-routing.md:178]: I wonder whether we should link to OData Connected Service and/or OData CLI here.

**Review by habbes — APPROVED** [review]
Looks good. Left minor comments for your consideration.

File: Odata-docs/webapi-8/fundamentals/metadata-routing.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    }
+}
+```
+
+---
+
+In the preceding block of code, we define two entity sets, namely, `Customers` and `Orders`. Implicitly, `Customer` and `Order` get included in the Edm model as entity types.
+
+## Controller
+We do not need to add any controller(s) to the handle service metadata requests. Requests for service metadata are handled by a built-in controller named `MetadataController`. The controller exposes two actions, namely:
```
Comment (ElizabethOkerio) [inline Odata-docs/webapi-8/fundamentals/metadata-routing.md:130]:
```suggestion
We do not need to add any controller(s) to the handle service metadata requests. Requests for service metadata are handled by a built-in controller named `MetadataController`. The controller exposes two actions, namely:
```
```suggestion
We do not need to add any controller(s) to handle service metadata requests. Requests for service metadata are handled by a built-in controller named `MetadataController`. The controller exposes two actions, namely:
```

Comment (ElizabethOkerio) [inline Odata-docs/webapi-8/fundamentals/metadata-routing.md:130]:
```suggestion
We do not need to add any controller(s) to handle service metadata requests. Requests for service metadata are handled by a built-in controller named `MetadataController`. The controller exposes two actions, namely:
```

File: Odata-docs/webapi-8/fundamentals/metadata-routing.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            "kind": "EntitySet",
+            "url": "Orders"
+        }
+    ]
+}
+```
+
+Based on the preceding payload, the entity sets in our OData service are `Customers` and `Orders`. The URLs for the two entity sets are `Customers` and `Orders` respectively.
+
+To retrieve all the customers, `Customers` would be appended to the service root to form the following URL:
```
Comment (ElizabethOkerio) [inline Odata-docs/webapi-8/fundamentals/metadata-routing.md:163]:
```suggestion
To retrieve all the customers, append `Customers` to the service root to form the following URL:
```

File: Odata-docs/webapi-8/fundamentals/metadata-routing.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+```http
+http://localhost:5000/odata/Customers
+```
+
+Similarly, to retrieve all the orders, `Orders` would be appended to the service root to form the following URL:
```
Comment (ElizabethOkerio) [inline Odata-docs/webapi-8/fundamentals/metadata-routing.md:168]:
```suggestion
Similarly, to retrieve all the orders, append `Orders` to the service root to form the following URL:
```

File: Odata-docs/webapi-8/fundamentals/metadata-routing.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        </Schema>
+    </edmx:DataServices>
+</edmx:Edmx>
+```
+
+Based on the preceding payload, two entity types are described within a shema element in the `MetadataRouting.Models` namespace: 
+- The `Customer` entity type contains two structural properties, namely, `Id` and `Name`, and a collection-valued navigation property named `Orders`. The `Id` property is designated as the key property.
+- The `Order` entity type contains two structural properties, namely, `Id` and `Amount`, and a single-valued navigation property named `Customer`. The `Id` property is designated as the key property.
+
+The payload further contains an `EntityContainer` element described within a schema element in the `Default` namespace. An OData service must define a entity container. Entity containers define the entity sets, singletons, function and action imports exposed by the OData service.
```
Comment (ElizabethOkerio) [inline Odata-docs/webapi-8/fundamentals/metadata-routing.md:228]:
```suggestion
The payload further contains an `EntityContainer` element described within a schema element in the `Default` namespace. An OData service must define an entity container. Entity containers define the entity sets, singletons, function and action imports exposed by the OData service.
```

**Review by ElizabethOkerio — APPROVED** [review]

**Pull request merged**